src/test/java/core/basesyntax/service/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,65 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertTrue;
+
+import core.basesyntax.database.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.handlers.BalanceOperationHandler;
+import core.basesyntax.strategy.handlers.OperationHandler;
+import core.basesyntax.strategy.handlers.PurchaseOperationHandler;
+import core.basesyntax.strategy.handlers.SupplyOperationHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitShopServiceImplTest {
+    private static String report;
+
+    @BeforeClass
+    public static void beforeClass() {
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap = new HashMap<>();
+        BalanceOperationHandler balanceOperationHandler = new BalanceOperationHandler();
+        SupplyOperationHandler supplyOperationHandler = new SupplyOperationHandler();
+        PurchaseOperationHandler purchaseOperationHandler = new PurchaseOperationHandler();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, balanceOperationHandler);
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY, supplyOperationHandler);
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE, purchaseOperationHandler);
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN, supplyOperationHandler);
+
+        List<String> fruitOperationsList = new ArrayList<>();
+        fruitOperationsList.add("type,fruit,quantity");
+        fruitOperationsList.add("b,banana,20");
+        fruitOperationsList.add("b,kiwi,10");
+        fruitOperationsList.add("b,apple,100");
+        fruitOperationsList.add("s,banana,100");
+        fruitOperationsList.add("s,peach,50");
+        fruitOperationsList.add("p,banana,13");
+        fruitOperationsList.add("r,apple,10");
+        fruitOperationsList.add("p,apple,20");
+
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlerMap);
+        FruitShopService fruitShopService = new FruitShopServiceImpl(operationStrategy);
+
+        TransactionParser transactionParser = new TransactionParserImpl();
+        List<FruitTransaction> fruitTransaction =
+                transactionParser.parseFruitTransactions(fruitOperationsList);
+
+        fruitShopService.provideOperation(fruitTransaction);
```

Comment:
you should test this service in this test

Reply:
When I am running tests with coverage it shows that this area covered with tests. It is exactly what I'm trying to test. Please tell me better way.